core/shared/src/main/scala/zio/internal/FiberContext.scala

```diff
@@ -128,10 +128,21 @@ private[zio] final class FiberContext[E, A](
   private[this] final def cutAncestryTrace(trace: ZTrace): ZTrace = {
     val maxExecLength  = platform.tracing.tracingConfig.ancestorExecutionTraceLength
     val maxStackLength = platform.tracing.tracingConfig.ancestorStackTraceLength
+    val maxAncestors   = platform.tracing.tracingConfig.ancestryLength - 1
 
-    trace.copy(
+    val cutParentTrace =
```

Comment:
This looks much better! Can we put `def truncateTrace(list: List[ZTrace], max: Int)` inside `ZTrace` companion object as well? Easier to test independently and this helps cut down on size of `FiberContext`.

Reply:
Done!